htdocs/admin/modules.php

```diff
@@ -230,13 +230,12 @@
 					// Now we install the module
 					if (!$error) {
 						@dol_delete_dir_recursive($dirins.'/'.$modulenameval); // delete the target directory
-						$submodulenamedir = $conf->admin->dir_temp.'/'.$tmpdir.'/htdocs/'.$modulenameval;
-						dol_syslog("We copy now directory ".$submodulenamedir." into target dir ".$dirins.'/'.$modulenameval);
-						$result = dolCopyDir($submodulenamedir, $dirins.'/'.$modulenameval, '0444', 1);
+						dol_syslog("We copy now directory ".$modulenamedir." into target dir ".$dirins.'/'.$modulenameval);
+						$result = dolCopyDir($modulenamedir, $dirins.'/'.$modulenameval, '0444', 1);
```

Comment:
We are in a loop that process $modulenameval
`foreach ($modulenamearrays as $modulenameval) {`
If you use $modulenamedir, because it has always the same value, you always copy the same dir.
If metapackage.conf contains
a
b
c
we must copy dir a into a , then b into b, then c into c.
With your change, source always has value a

Reply:
You're right, I did not take into account the metapackages, the change has been made and tested.